revelatormodsystem.py

```diff
@@ -2,49 +2,67 @@
 import sys
 import os
 
-scriptextract = False
+script_extract_folder = ''
 basepath = ''
+mod_folder = ''
+
+
 def bytearrtomessage(barr):
-	result = ""	
+	result = ""
 	for b in barr:
 		result += hex(b) + ','
 	print result[:4]
 	return result[:-1]
+
+
 def on_message(message, data):
-	global script
-	global scriptextract
-	global basepath
 	print message
-	currscriptpath = os.path.join(basepath, 'ggmods', message['payload'] + '.ggscript');
-	if scriptextract and not os.path.isfile(os.path.join(basepath, 'originalscripts', message['payload'] + '.ggscript')):
-		origscript = open(os.path.join(basepath, 'originalscripts', message['payload'] + '.ggscript'), 'wb')
-		origscript.write(bytearray(data))	
-		origscript.close()
-	elif scriptextract:
-		print 'originalscripts/' + message['payload'] + '.ggscript already exists.  Not overwriting.'
+	script_name = message['payload'] + '.ggscript'
+
+	# Copy the script into the extract folder if needed
+	if script_extract_folder:
+		original_script_path = os.path.join(script_extract_folder, script_name)
+		if not os.path.isfile(original_script_path):
+			origscript = open(original_script_path, 'wb')
+			origscript.write(bytearray(data))
+			origscript.close()
+		else:
+			print original_script_path + ' already exists.  Not overwriting.'
+
+	# post the received data
+	currscriptpath = os.path.join(mod_folder, script_name)
 	if(os.path.isfile(currscriptpath)):
-		print 'Loading ' + message['payload'] + '.ggscript'
+		print 'Loading ' + script_name
 		scriptfile = open(currscriptpath, 'rb')
 		messagepayload = bytearrtomessage(bytearray(scriptfile.read()))
 		scriptfile.close()
 		script.post({'payload': messagepayload})
 	else:
 		print currscriptpath + " not found.  Either you don't have a mod for that character, the ETC file wasn't required, or you misspelled a file name."
 		script.post({'type':str(2), 'payload': ""})
+
+
 if __name__ == '__main__':
-	global basepath

```

Comment:
Useless global here since you're in the same context as the variable declaration.